inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/KubernetesFileReader.java

```diff
@@ -92,11 +111,36 @@ public PodList getPods() {
     /**
      * get pod metadata by namespace and pod name
      */
-    public ObjectMeta getPodMetadata(String namespace, String podName) {
-        List<ObjectMeta> objectMetas = client.pods().list().getItems().stream().map(Pod::getMetadata)
-                .filter(data -> data.getName().equalsIgnoreCase(podName) && data.getNamespace()
-                        .equalsIgnoreCase(namespace)).collect(Collectors.toList());
-        return CollectionUtils.isNotEmpty(objectMetas) ? objectMetas.get(0) : null;
+    public Map<String, String> getK8sMetadata(JobProfile jobConf) {
+        if (Objects.isNull(jobConf)) {
+            return null;
+        }
+        Map<String, String> k8sInfo = MetaDataUtils.getLogInfo(fileReaderOperator.file.getName());
+        if (k8sInfo.isEmpty()) {
+            return null;
+        }
+        List<String> namespaces = MetaDataUtils.getNamespace(jobConf);
+        if (Objects.isNull(namespaces) || namespaces.isEmpty()) {
+            return null;
+        }
+        if (!namespaces.contains(k8sInfo.get(NAMESPACE))) {
+            return null;
+        }
+        Pod pod = client.pods().inNamespace(k8sInfo.get(NAMESPACE)).withName(k8sInfo.get(POD_NAME)).get();
+        PodList podList = client.pods().inNamespace(k8sInfo.get(NAMESPACE))
+                .withLabels(MetaDataUtils.getPodLabels(jobConf)).list();
+        Gson gson = new Gson();
```

Comment:
It is recommended to use a global static Gson object.

Reply:
done